wemix/bind/backends/wemix_simulated_test.go

```diff
@@ -0,0 +1,175 @@
+package backends_test
+
+import (
+	"context"
+	"encoding/hex"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+	gov "github.com/ethereum/go-ethereum/wemix/bind"
+	"github.com/ethereum/go-ethereum/wemix/bind/backends"
+	wemix_miner "github.com/ethereum/go-ethereum/wemix/miner"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWemixBackends(t *testing.T) {
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+	opts, err := bind.NewKeyedTransactorWithChainID(key, backends.ChainID)
+	require.NoError(t, err)
+	b, err := backends.NewWemixSimulatedBackend(
+		key,
+		t.TempDir(),
+		core.GenesisAlloc{
+			opts.From: {Balance: new(big.Int).Mul(big.NewInt(256_000_000_000_000), big.NewInt(params.Ether))},
+		},
+		backends.SetLogLevel(2),
+	)
+	if err != nil {
+		return
+	}
+	require.NotNil(t, b)
+
+	defaultEnv := gov.DefaultInitEnvStorage
+
+	require.True(t, defaultEnv.MAX_PRIORITY_FEE_PER_GAS.Cmp(wemix_miner.SuggestGasPrice()) == 0)
+	coinBase, err := wemix_miner.GetCoinbase(common.Big0)
+	require.NoError(t, err)
+	require.Equal(t, opts.From, coinBase)
+
+	callOpts := new(bind.CallOpts)
+	contracts, err := gov.GetGovContractsByOwner(callOpts, b, opts.From)
+	require.NoError(t, err)
+
+	GetBallotDurationMax, err := contracts.EnvStorageImp.GetBallotDurationMax(callOpts)
+	require.NoError(t, err)
+	require.True(t, defaultEnv.BALLOT_DURATION_MAX.Cmp(GetBallotDurationMax) == 0)
```

Comment:
It is recommended to modify it in the following manner.
1. improved variable names.
2. print a message in case of failure.
3. modify codes with similar patterns as well.
```suggestion
	got, err := contracts.EnvStorageImp.GetBallotDurationMax(callOpts)
	require.NoError(t, err)
	require.Truef(t, defaultEnv.BALLOT_DURATION_MAX.Cmp(got) == 0, "mismatch BALLOT_DURATION_MAX, got: %v, want: %v", got, defaultEnv.BALLOT_DURATION_MAX)
```